gnucash/register/ledger-core/split-register-model.c

```diff
@@ -1389,8 +1389,8 @@ gnc_split_register_get_price_entry (VirtualLocation virt_loc,
 
     split = gnc_split_register_get_split (reg, virt_loc.vcell_loc);
 
-    amount = xaccSplitGetAmount (split);
-    if (gnc_numeric_zero_p (amount))
+    if (gnc_numeric_zero_p (xaccSplitGetAmount(split)) ||
+        gnc_numeric_zero_p (xaccSplitGetValue(split)))
         return NULL;
```

Comment:
With this change the test of `gnc_numeric_zero_p(price)` below is redundant and should be removed.